### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by dtebbs**

### CODE REVIEW

**Comment by AntoineRondelet** [comment]
What about using https://github.com/scipr-lab/libff/blob/04094606d6e90f4d6a2d6a9d32a3a5a660b67bd4/libff/common/utils.hpp#L43 instead of void casting?

File: libsnark/relations/constraint_satisfaction_problems/r1cs/r1cs.tcc [hunk 1]
```diff
@@ -175,6 +175,7 @@ void r1cs_constraint_system<FieldT>::add_constraint(const r1cs_constraint<FieldT
 #ifdef DEBUG
     constraint_annotations[constraints.size()] = annotation;
 #endif
+    (void)annotation;
```
Comment (AntoineRondelet) [inline libsnark/relations/constraint_satisfaction_problems/r1cs/r1cs.tcc]: Let's put this one in an `#else`
Something like:
```
#ifdef DEBUG
    constraint_annotations[constraints.size()] = annotation;
#else
    libff::UNUSED(annotation);
#endif
```

File: libsnark/gadgetlib1/protoboard.tcc [hunk 2]
```diff
@@ -116,6 +116,8 @@ void protoboard<FieldT>::augment_variable_annotation(const pb_variable<FieldT> &
     auto it = constraint_system.variable_annotations.find(v.index);
     constraint_system.variable_annotations[v.index] = (it == constraint_system.variable_annotations.end() ? "" : it->second + " ") + postfix;
 #endif
+    (void)v;
```
Comment (AntoineRondelet) [inline libsnark/gadgetlib1/protoboard.tcc]: Move into a `#else` to have something like
```
#ifdef DEBUG
    <if block>;
#else
    libff::UNUSED(v);
    libff::UNUSED(postfix);
#endif
```

Comment (AntoineRondelet) [reply libsnark/gadgetlib1/protoboard.tcc]: Let's follow a similar approach for the other places where you "void casted" in order to be consistent with unused variable handling.

File: libsnark/relations/constraint_satisfaction_problems/r1cs/r1cs.tcc (hunk 1 above)
Comment (dtebbs) [reply libsnark/relations/constraint_satisfaction_problems/r1cs/r1cs.tcc]: Thanks.  I hadn't noticed the `libff::UNUSED`.  Switched to use that everywhere.

File: libsnark/gadgetlib1/protoboard.tcc (hunk 2 above)
Comment (dtebbs) [reply libsnark/gadgetlib1/protoboard.tcc]: OK.  I've tried to add the `#ifdef` blocks in.  Sometimes, they use `DEBUG` in the, in which case I do exactly as in your comment.  In some cases, we have to use `NDEBUG` in the conditional, where a variable is only defined to be used in an assert.

**Comment by AntoineRondelet** [comment]
LGTM

**Pull request merged**